### PR TITLE
use zephyr's build-in tinycbor for serial recovery build

### PR DESCRIPTION
**Pull request by nvlsianpu**
Zephyr has build-in tinycbor code of version which doesn't
require newlib-c by default. Also the cborattr module is now
taken form the zephyr code.
This patch decreases flash consumption by 10 KB for serial recovery build.

fix default value for serial recovery mode detect port for nRFx devices .
Zephyr's SOC_FAMILY_NRF5 keyword was changed to SOC_FAMILY_NRF to
align with zephyr code-base.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: boot/boot_serial/src/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,8 @@
 zephyr_sources(boot_serial.c)
 
 zephyr_include_directories(${BOOT_DIR}/bootutil/include)
+
+zephyr_include_directories($ENV{ZEPHYR_BASE}/ext/lib/mgmt/mcumgr/cborattr/include)
```
Comment (nvlsianpu) [inline boot/boot_serial/src/CMakeLists.txt]: @carlescufi I'm aware of that cborattr should be extracted to a zephyr module. I can do this after this patch  (which I prefer).

File: boot/boot_serial/src/boot_serial.c [hunk 2]
```diff
@@ -216,7 +217,8 @@ bs_upload(char *buf, int len)
 
     memset(img_data, 0, sizeof(img_data));
     cbor_buf_reader_init(&reader, (uint8_t *)buf, len);
-    cbor_parser_init(&reader.r, 0, &parser, &value);
+    //cbor_parser_init(&reader.r, 0, &parser, &value);
```
Comment (utzig) [inline boot/boot_serial/src/boot_serial.c]: Why was this commented out and not removed?

File: boot/boot_serial/src/boot_serial.c [hunk 3]
```diff
@@ -326,7 +328,8 @@ boot_serial_input(char *buf, int len)
     buf += sizeof(*hdr);
     len -= sizeof(*hdr);
     bs_writer.bytes_written = 0;
-    cbor_encoder_init(&bs_root, &bs_writer, 0);
+    //cbor_encoder_init(&bs_root, &bs_writer, 0);
```
Comment (utzig) [inline boot/boot_serial/src/boot_serial.c]: Same...

File: boot/boot_serial/src/boot_serial.c (hunk 2 above)
Comment (nvlsianpu) [reply boot/boot_serial/src/boot_serial.c]: obviously I mess with Mynewt here, thx for this comment!

**Comment by nvlsianpu** [comment]
@utzig Updated!

**Pull request merged**